api/sendfild/main.go

```diff
@@ -0,0 +1,202 @@
+	"github.com/spf13/viper"
+	"github.com/textileio/go-threads/util"
+	"github.com/textileio/powergate/v2/lotus"
+	"github.com/textileio/textile/v2/api/sendfild/service"
+	"github.com/textileio/textile/v2/cmd"
+)
+
+const daemonName = "sendfild"
+
+var (
+	log = logging.Logger(daemonName)
+
+	config = &cmd.Config{
+		Viper: viper.New(),
+		Dir:   "." + daemonName,
+		Name:  "config",
+		Flags: map[string]cmd.Flag{
+			"debug": {
+				Key:      "debug",
+				DefValue: false,
+			},
+			"logFile": {
+				Key:      "log_file",
+				DefValue: "", // no log file
+			},
+			"listenAddr": {
+				Key:      "listen_addr",
+				DefValue: "127.0.0.1:5000",
+			},
+			"lotusAddr": {
+				Key:      "lotus_addr",
+				DefValue: "127.0.0.1:7777",
+			},
+			"lotusAuthToken": {
+				Key:      "lotus_auth_token",
+				DefValue: "",
+			},
+			"lotusConnRetries": {
+				Key:      "lotus_conn_retries",
+				DefValue: 2,
+			},
+			"mongoUri": {
+				Key:      "mongo_uri",
+				DefValue: "mongodb://127.0.0.1:27017",
+			},
+			"mongoDb": {
+				Key:      "mongo_db",
+				DefValue: "textile_sendfil",
+			},
+			"messageWaitTimeout": {
+				Key:      "message_wait_timeout",
+				DefValue: time.Minute * 5,
+			},
+			"messageConfidence": {
+				Key:      "message_confidence",
+				DefValue: uint64(2),
+			},
+		},
+		EnvPre: "SENDFIL",
+		Global: true,
+	}
+)
+
+func init() {
+	cobra.OnInitialize(cmd.InitConfig(config))
+	cmd.InitConfigCmd(rootCmd, config.Viper, config.Dir)
+
+	rootCmd.PersistentFlags().StringVar(
+		&config.File,
+		"config",
+		"",
+		"Config file (default ${HOME}/"+config.Dir+"/"+config.Name+".yml)")
+	rootCmd.PersistentFlags().BoolP(
+		"debug",
+		"d",
+		config.Flags["debug"].DefValue.(bool),
+		"Enable debug logging")
+	rootCmd.PersistentFlags().String(
+		"logFile",
+		config.Flags["logFile"].DefValue.(string),
+		"Write logs to file")
+
+	rootCmd.PersistentFlags().String(
+		"listenAddr",
+		config.Flags["listenAddr"].DefValue.(string),
+		"Sendfil API listen address")
+
+	rootCmd.PersistentFlags().String(
+		"lotusAddr",
+		config.Flags["lotusAddr"].DefValue.(string),
+		"Lotus API address")
+	rootCmd.PersistentFlags().String(
+		"lotusAuthToken",
+		config.Flags["lotusAuthToken"].DefValue.(string),
+		"Lotus API auth token")
+	rootCmd.PersistentFlags().Int(
+		"lotusConnRetries",
+		config.Flags["lotusConnRetries"].DefValue.(int),
+		"Lotus API connection retry count")
+
+	rootCmd.PersistentFlags().String(
+		"mongoUri",
+		config.Flags["mongoUri"].DefValue.(string),
+		"MongoDB connection URI")
+	rootCmd.PersistentFlags().String(
+		"mongoDb",
+		config.Flags["mongoDb"].DefValue.(string),
+		"MongoDB database name")
+
+	rootCmd.PersistentFlags().Duration(
+		"messageWaitTimeout",
+		config.Flags["messageWaitTimeout"].DefValue.(time.Duration),
+		"Timeout for listening for messages to become active on chain")
+
+	rootCmd.PersistentFlags().Int64(
+		"messageConfidence",
+		config.Flags["messageConfidence"].DefValue.(int64),
```

Comment:
Getting a panic here about this flag.
```
$ docker run --rm textile/sendfil:staging
panic: interface conversion: interface {} is uint64, not int64

goroutine 1 [running]:
main.init.0()
        /textile/api/sendfild/main.go:128 +0xc0c
```

Think it's referring to these not matching? https://github.com/textileio/textile/pull/512/files?file-filters%5B%5D=.go&file-filters%5B%5D=.mod&file-filters%5B%5D=.proto&file-filters%5B%5D=.sum#diff-af38636851147b7aa79fdbf910e023fecb38fd9fe6826cff5e2c21111e26da27R67

Reply:
Fixed